core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -75,7 +77,7 @@ private[spark] class BlockManager(
 
   val diskBlockManager = new DiskBlockManager(this, conf)
 
-  private val blockInfo = new TimeStampedHashMap[BlockId, BlockInfo]
+  private val blockInfo = new ConcurrentHashMap[BlockId, BlockInfo]
```

Comment:
I was worried about the pitfalls of `.getOrElseUpdate` not being atomic on `ConcurrentHashMaps` that had been wrapped into Scala maps.